datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -118,6 +118,8 @@ pub enum LogicalPlan {
     SetVariable(SetVariable),
     /// Prepare a statement
     Prepare(Prepare),
+    /// Insert / Update / Delete
+    Write(WriteRel),
```

Comment:
The goal of the PR

Reply:
I wonder if calling this Dml(DmlStatement) rather than `Write` might be a more standard term? However, I think this term this may come from Substrait: https://substrait.io/relations/logical_relations/#write-operator so that is ok too. 

In general I would like to split logical plan up a bit more into "relational operators" and "non-relational actions" (like SetVariable, for example)